packages/rum-core/test/common/apm-server.spec.js

```diff
@@ -119,38 +119,22 @@ describe('ApmServer', function() {
     )
   })
 
-  xit('should queue items', function() {
-    spyOn(loggingService, 'warn').and.callThrough()
+  it('should queue items before sending', function() {
     configService.setConfig({
-      serviceName: 'serviceName',
-      throttlingRequestLimit: 1
+      serviceName: 'serviceName'
     })
-    spyOn(apmServer, '_postJson').and.callThrough()
-    spyOn(apmServer, '_makeHttpRequest').and.callThrough()
+
+    spyOn(apmServer, '_postJson')
     apmServer.init()
-    spyOn(apmServer, '_throttledMakeRequest').and.callThrough()
 
-    var trs = generateTransaction(19)
+    let trs = generateTransaction(19)
     trs.forEach(apmServer.addTransaction.bind(apmServer))
     expect(apmServer.transactionQueue.items.length).toBe(19)
     expect(apmServer._postJson).not.toHaveBeenCalled()
-    trs = generateTransaction(1)
-    trs.forEach(apmServer.addTransaction.bind(apmServer))
 
+    apmServer.transactionQueue.flush()
```

Comment:
if we do this, we are sending transactions to the apm server. I feel like we are leaking here. Can you make it async and await for the response from apm server?